simpledb/record/table_scan.go

```diff
@@ -0,0 +1,175 @@
+package record
+
+import (
+	"simpledb/file"
+	"simpledb/tx"
+)
+
+type TableScan struct {
+	tx          *tx.Transaction
+	layout      *Layout
+	rp          *RecordPage
+	filename    string
+	currentSlot int32
+}
+
+// 本当はUpdateScanのインターフェースを満たすべきだが、UpdateScanのメソッドは未実装なので今は無視
+func NewTableScan(tx *tx.Transaction, tableName string, layout *Layout) (*TableScan, error) {
+	tableScan := &TableScan{tx, layout, nil, tableName + ".tbl", -1}
+	size, err := tx.Size(tableScan.filename)
+	if err != nil {
+		return nil, err
+	}
+	if size == 0 {
+		err = tableScan.moveToNewBlock()
+	} else {
+		tableScan.moveToBlock(0)
+	}
+	if err != nil {
+		return nil, err
+	}
+	return tableScan, nil
+}
+
+func (ts *TableScan) BeforeFirst() {
+	ts.moveToBlock(0)
+}
+
+func (ts *TableScan) Next() (bool, error) {
+	var err error
+	ts.currentSlot, err = ts.rp.NextAfter(ts.currentSlot)
+	if err != nil {
+		return false, err
+	}
+	for ts.currentSlot < 0 {
+		atLastBlock, err := ts.atLastBlock()
+		if err != nil {
+			return false, err
+		}
+		if atLastBlock {
+			return false, nil
+		} else {
+			ts.moveToBlock(ts.rp.Block().Number + 1)
+			ts.currentSlot, err = ts.rp.NextAfter(ts.currentSlot)
+			if err != nil {
+				return false, err
+			}
+		}
```

Comment:
early returnしてたらelseはつけないのがgo-wayな気がします

```suggestion
		if atLastBlock {
			return false, nil
		}
		ts.moveToBlock(ts.rp.Block().Number + 1)
		ts.currentSlot, err = ts.rp.NextAfter(ts.currentSlot)
		if err != nil {
			return false, err
		}
```